File: api/schemas/agent.py
from pydantic import BaseModel
from typing import Optional, List


class ChatMessage(BaseModel):
    content: str
    session_id: Optional[str] = None


class ChatResponse(BaseModel):
    message_id: str
    content: str
    tools_used: List[str] = []
    usage: dict = {}


class ToolCallInfo(BaseModel):
    tool_name: str
    arguments: dict
    result: Optional[dict] = None

File: archive/frontend/src/archive/agent/agent.ts
import { get, post } from './index'

export interface ChatMessage {
  content: string
  session_id?: string
}

export interface ChatResponse {
  message_id: string
  content: string
  tools_used: string[]
  usage: Record<string, number>
}

export interface ToolCallInfo {
  tool_name: string
  arguments: Record<string, any>
  result?: Record<string, any>
}

export const agentApi = {
  sendMessage: (data: ChatMessage) =>
    post<ChatResponse>('/chat', data),

  getHistory: (sessionId: string) =>
    get<ChatMessage[]>(`/chat/history/${sessionId}`),
}
